### ISSUE TITLE
1.2.161 regression with typed arrow function

### ISSUE THREAD

**Issue by dpchamps** [swc-project/swc#4178]
### Describe the bug

Bumping to 1.2.161 has surfaced a regression, which causes the compiler to throw for valid input in a specific context.

### Input code

```typescript
export const x = (a: number) => {
  const y = a && b ? (z: number): string => "hello" : null
}
```

### Config

```json
{
  "sourceMaps": true,
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "transform": {
      "react": {
        "runtime": "automatic"
      }
    },
    "target": "es2022"
  },
  "minify": false
}
```

### Playground link

https://play.swc.rs/?version=1.2.161&code=H4sIAAAAAAAAA0utKMgvKlFIzs8rLlGoULBV0Ei0UsgrzU1KLdJUsLVTqOZSgEpWAiUTFdTUFJIU7BU0quCqrBSKS4oy89JBqpUyUnNy8pUUQJI5OVy1AGbPJwleAAAA&config=H4sIAAAAAAAAAyWLSwoDIRAF7%2FLWswguvUMO0UgbDOOH7hZGBu8eneyKouqG1i6B39QU3qTzga8G%2BBuNRFk26ShGFzxsNNYgqRkOmF7%2FYy4WKhqr5J0LU7AHerGUeY3UrWayFDB3TfLhVYDVvZzDUjmVFAd8pFN5%2FgCKU2KVlgAAAA%3D%3D

### Expected behavior

For the compiler to emit code as expected

### Actual behavior

The compiler throws, looks the context expects an lambda token instead of a type.

```

  [31mx[0m Expected '=>', got ':'
   ,-[[36;1;4minput.tsx[0m:2:3]
 [2m2[0m | const y = a && b ? (z: number): string => "hello" : null
   : [31;1m                              ^[0m
   `----
```

### Version

1.2.161

**Comment by williamtetlow** [comment]
I can take a look at this

**Comment by dpchamps** [comment]
@kdy1 @williamtetlow this regression was only partially fixed in #4199 

[See .163 example failing](https://play.swc.rs/?version=1.2.163&code=H4sIAAAAAAAAA0utKMgvKlFIzs8rLlGoULBV0Ei0UsgrzU1KLdJUsLVT0ACT1VwKUCWVQCWJCmpqCkkK9goaVXC1VgrFJUWZeekg1UoZqTk5%2BUoKIMmcHK5aAHl1v5dkAAAA&config=H4sIAAAAAAAAAyWLSwoDIRAF7%2FLWswguvUMO0UgbDOOH7hZGBu8eneyKouqG1i6B39QU3qTzga8G%2BBuNRFk26ShGFzxsNNYgqRkOmF7%2FYy4WKhqr5J0LU7AHerGUeY3UrWayFDB3TfLhVYDVvZzDUjmVFAd8pFN5%2FgCKU2KVlgAAAA%3D%3D)

[Compared to .160 example passing](https://play.swc.rs/?version=1.2.160&code=H4sIAAAAAAAAA0utKMgvKlFIzs8rLlGoULBV0Ei0UsgrzU1KLdJUsLVT0ACT1VwKUCWVQCWJCmpqCkkK9goaVXC1VgrFJUWZeekg1UoZqTk5%2BUoKIMmcHK5aAHl1v5dkAAAA&config=H4sIAAAAAAAAAyWLSwoDIRAF7%2FLWswguvUMO0UgbDOOH7hZGBu8eneyKouqG1i6B39QU3qTzga8G%2BBuNRFk26ShGFzxsNNYgqRkOmF7%2FYy4WKhqr5J0LU7AHerGUeY3UrWayFDB3TfLhVYDVvZzDUjmVFAd8pFN5%2FgCKU2KVlgAAAA%3D%3D)

**Comment by williamtetlow** [comment]
@dpchamps thanks for finding this!! I'd checked an arrow function returned from function worked but not an arrow 🤦 

fix has been merged, see [`1.2.164`](https://play.swc.rs/?version=1.2.164&code=H4sIAAAAAAAAA0utKMgvKlFIzs8rLlGoULBV0Ei0UsgrzU1KLdJUsLVT0ACT1VwKQABRVAlUlKigpqaQpGCvoFEFV22lUFxSlJmXDlKvlJGak5OvpACSzMmx5qq15gIAHkJWCGkAAAA%3D&config=H4sIAAAAAAAAAzWMSwoDMQxD76L1LEqWuUMPYYxTUpoPtgMThrl7k0J30uNJF6wNZXlSN0TXIQfexogXOqmJ7mSzOp2I8NnFWHN3HHBbKNHH5F5FqVpqWravQuy%2FMKrnImtJw1shz4x726QvWQbEwiMELFRyzWn%2BD78yPC5plwAAAA%3D%3D)
